### PR TITLE
[12.0] [MIG] web_widget_x2many_2d_matrix_example

### PR DESCRIPTION
**Pull request by anandkansagra**
Issue [Reference](https://github.com/OCA/web/issues/1048)

### CODE REVIEW

File: web_widget_x2many_2d_matrix_example/models/x2m_demo.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            'view_mode': 'form',
+            'res_model': 'x2m.matrix.demo.wiz',
+            'target': 'new',
+            'res_id': wiz.id,
+            'context': self.env.context,
+        }
+
+
+class X2MDemoLine(models.Model):
+    _name = 'x2m.demo.line'
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/models/x2m_demo.py:28]:
```
2018-12-31 06:19:47,578 152 WARNING openerp_test odoo.models: The model x2m.demo.line has no _description
```

File: web_widget_x2many_2d_matrix_example/models/x2m_demo.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models, api, fields
+
+
+class X2MDemo(models.Model):
+    _name = 'x2m.demo'
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/models/x2m_demo.py:6]:
```
2018-12-31 06:19:47,578 152 WARNING openerp_test odoo.models: The model x2m.demo has no _description
```

File: web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class X2mMatrixDemoWiz(models.TransientModel):
+    _name = 'x2m.matrix.demo.wiz'
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.py:6]:
```
2018-12-31 06:19:47,578 152 WARNING openerp_test odoo.models: The model x2m.matrix.demo.wiz has no _description
```

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.py (hunk 3 above)
Comment (anandkansagra) [reply web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.py:6]: Done @alexey-pelykh , Now runbot seems green

File: web_widget_x2many_2d_matrix_example/views/x2m_demo.xml [hunk 4]
```diff
@@ -1,54 +1,53 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
-  <record id="view_x2m_demo_form" model="ir.ui.view">
+    <record id="view_x2m_demo_form" model="ir.ui.view">
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/views/x2m_demo.xml:4]: Bad indent

File: web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.xml [hunk 5]
```diff
@@ -1,21 +1,22 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
 
-  <record id="x2many_2d_matrix_demo" model="ir.ui.view">
+    <record id="x2many_2d_matrix_demo" model="ir.ui.view">
     <field name="name">x2m.matrix.demo.wiz</field>
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/wizard/x2m_matrix.xml]: Bad indent

File: web_widget_x2many_2d_matrix_example/README.rst [hunk 6]
```diff
@@ -23,7 +23,7 @@ Contributors
 ------------
 
 * Simone Orsi <simone.orsi@camptocamp.com>
-
+* Anand Kansagra <kansagraanand@hotmail.com>
```
Comment (alexey-pelykh) [inline web_widget_x2many_2d_matrix_example/README.rst:26]: Please split README.rst as per https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: web_widget_x2many_2d_matrix_example/README.rst (hunk 6 above)
Comment (anandkansagra) [reply web_widget_x2many_2d_matrix_example/README.rst:26]: Done

**Review by tarteo — APPROVED** [review]
Tested. It looks like the total cell is not working. But that's an issue of the widget itself I think. I'll make an issue for this
image

**Comment by alexey-pelykh** [comment]
#1174 should address the total field issue

**Review by alexey-pelykh — APPROVED** [review]
Looks good!

**Review by simahawk — APPROVED** [review]
next time, pls, either you replace README.rst w/ the fake one or you run [readme generator](https://github.com/OCA/maintainer-tools#readme-generator) to get the final version ;)

**Pull request merged**